Lib/test/test_structseq.py

```diff
@@ -133,6 +135,74 @@ def test_match_args_with_unnamed_fields(self):
         self.assertEqual(os.stat_result.n_unnamed_fields, 3)
         self.assertEqual(os.stat_result.__match_args__, expected_args)
 
+    def test_copy_replace_all_fields_visible(self):
+        assert os.times_result.n_unnamed_fields == 0 and os.times_result.n_sequence_fields == os.times_result.n_fields
+
+        expected_args = ('user', 'system', 'children_user', 'children_system', 'elapsed')
+        self.assertEqual(os.times_result.__match_args__, expected_args)
+        self.assertEqual(os.times_result.n_fields, len(expected_args))
+
+        n_fields = os.times_result.n_fields  # 5
+        t = os.times_result(range(os.times_result.n_fields))
+
+        # visible fields
+        self.assertEqual(copy.replace(t), tuple(range(n_fields)))
+        self.assertEqual(copy.replace(t, user=1), (1, *range(1, n_fields)))
+        self.assertEqual(copy.replace(t, system=2), (0, 2, *range(2, n_fields)))
+        self.assertEqual(copy.replace(t, user=1, system=2), (1, 2, *range(2, n_fields)))
+
+        # unknown fields
+        with self.assertRaisesRegex(TypeError, 'unexpected field name'):
+            copy.replace(t, error=-1)
+        with self.assertRaisesRegex(TypeError, 'unexpected field name'):
+            copy.replace(t, user=1, error=-1)
+
+    def test_copy_replace_with_invisible_fields(self):
+        t = time.gmtime(0)
+        assert t.n_unnamed_fields == 0 and t.n_sequence_fields < t.n_fields
+
+        # visible fields
+        self.assertEqual(copy.replace(t), (1970, 1, 1, 0, 0, 0, 3, 1, 0))
```

Comment:
Please test also the type and "invisible" attributes of the result.
```suggestion
        t2 = copy.replace(t)
        self.assertEqual(t2, (1970, 1, 1, 0, 0, 0, 3, 1, 0))
        self.assertIsInstance(t2, time.struct_time)
        self.assertEqual(t2.tm_zone, 'GMT')
```